earthfile2llb/with_docker_run_reg.go

```diff
@@ -126,7 +129,8 @@ func (w *withDockerRunRegistry) Run(ctx context.Context, args []string, opt With
 	}
 
 	defer func() {
-		cmd.SetEndError(retErr)
```

Comment:
I wonder if you could keep this existing function and instead wrap the error? e.g. `errors.Wrap(retErr, "WITH DOCKER RUN")`?